tools/halide_trace_config.h

```diff
@@ -101,22 +103,104 @@ struct Label {
     }
 };
 
-// Configuration for how a func should be rendered in HalideTraceViz
+// Configuration for how a func should be rendered in HalideTraceViz.
 struct FuncConfig {
-    float zoom = 1.f;
-    int load_cost = 0;
-    int store_cost = 1;
-    Point pos;
-    std::vector<Point> strides = { {1, 0}, {0, 1} };
-    int color_dim = -1;
-    float min = 0.f, max = 1.f;
+    // Note that every field in this struct is initialized to a value which
+    // means "no value specified"; this is allows us to merge configs
+    // from several sources (auto-layout, embedded trace-tags, and command-line)
+    // in a way that we can selectively add or override some-but-not-all configuration
+    // values (e.g., use auto-layout's positioning, but customizing labels
+    // and changing rgb vs gray rendering). In all cases, if a field is
+    // the initial "no value specified" value at rendering time, HalideTraceViz
+    // will choose a reasonable value for that field.
+
+    // Each value of a Func will draw as a zoom x zoom
+    // box in the output. Fractional values are allowed.
+    //
+    // Valid values: 0.0 < zoom <= HUGEVAL or so
+    float zoom = -1.f;
+
+    // Each load from a Func costs the given number of ticks.
+    // Legal values are 0.0 < zoom <= 1000 or so
+    //
+    // Valid values: load_cost >= 0
+    int load_cost = -1;
+
+    // Each store to a Func costs the given number of ticks.
+    //
+    // Valid values: store_cost >= 0
+    int store_cost = -1;
+
+    // The position on the screen corresponding to the Func's 0, 0 coordinate.
+    //
+    // Valid values: pos.x >= 0, pos.y >= 0
+    Point pos = { -1, -1 };
```

Comment:
If a Func is realized over 100, 100 -> 200, 200, then you may well want to do use -1. Maybe INT_MIN instead?